tosfs/core.py

```diff
@@ -413,6 +416,200 @@ def _try_dir_info(self, bucket: str, key: str, path: str, fullpath: str) -> dict
         except Exception as e:
             raise TosfsError(f"Tosfs failed with unknown error: {e}") from e
 
+    def exists(self, path: str, **kwargs: Union[str, bool, float, None]) -> bool:
+        """Check if a path exists in the TOS file system.
+
+        Parameters
+        ----------
+        path : str
+            The path to check for existence.
+        **kwargs : dict, optional
+            Additional arguments.
+
+        Returns
+        -------
+        bool
+            True if the path exists, False otherwise.
+
+        Raises
+        ------
+        tos.exceptions.TosClientError
+            If there is a client error while checking the path.
+        tos.exceptions.TosServerError
+            If there is a server error while checking the path.
+        TosfsError
+            If there is an unknown error while checking the path.
+
+        Examples
+        --------
+        >>> fs = TosFileSystem()
+        >>> fs.exists("tos://bucket/to/file")
+        True
+        >>> fs.exists("tos://mybucket/nonexistentfile")
+        False
+
+        """
+        if path in ["", "/"]:
+            # the root always exists, even if anon
+            return True
+
+        path = self._strip_protocol(path)
+        bucket, key, version_id = self._split_path(path)
+        # if the path is a bucket
+        if not key:
+            return self._exists_bucket(bucket)
+        else:
+            object_exists = self._exists_object(bucket, key, path, version_id)
+            if not object_exists:
+                return self._exists_object(
+                    bucket, key.rstrip("/") + "/", path, version_id
+                )
+            return object_exists
+
+    def _exists_bucket(self, bucket: str) -> bool:
+        """Check if a bucket exists in the TOS file system.
+
+        It will first check the dircache,
+        then check the bucket using the TOS client.
+
+        Parameters
+        ----------
+        bucket : str
+            The name of the bucket to check for existence.
+
+        Returns
+        -------
+        bool
+            True if the bucket exists, False otherwise.
+
+        Raises
+        ------
+        tos.exceptions.TosClientError
+            If there is a client error while checking the bucket.
+        tos.exceptions.TosServerError
+            If there is a server error while checking the bucket.
+        TosfsError
+            If there is an unknown error while checking the bucket.
+
+        Examples
+        --------
+        >>> fs = TosFileSystem()
+        >>> fs._exists_bucket("mybucket")
+        True
+        >>> fs._exists_bucket("nonexistentbucket")
+        False
+
+        """
+        if self.dircache.get(bucket, False):
+            return True
+        else:
+            try:
+                if self._ls_from_cache(bucket):
```

Comment:
ls_from_cache ? which means we will get a false positive result when accessing the the cache ?

Reply:
I mean if we've already removed the object key from backend object storage, while the cache is still holding the cached item, then we will regard the object key is still exising, which is incorrect ?

Reply:
Yes, there may be multi-source consistency issues here. The current cache can only guarantee that the cached data is invalidated when it performs the deletion operation by itself. To ensure accuracy, one way is not to adopt the caching mechanism or to pass the refresh parameter everywhere. WDYT?

Reply:
After some thinking, I feel that the disadvantages of introducing the list cache outweigh the advantages for us. The main considerations are as follows: 

* We cannot assume that users read and write the TOS only through the single entry of TOSFS. The cost of maintaining cache consistency and the cost of misjudgment may be more serious compared to performance. 
* If we assume that the main usage scenario of the target users is programming, then FS-level APIs such as those dependent on caching like ls, exists, info... will not be used so frequently. 

After seeing that fsspec has built-in basic support for `dircache` and s3fs followed suit, I also introduced caching by default. But now I feel that not introducing caching seems to be a better option.  WDYT? @openinx @xianyinxin

Reply:
@openinx After discussing with @xianyinxin , we agreed with removing cache logic from tosfs, the biggest concern is for some dangerous operations(e.g. `rm`), if misjudgments are caused by caching, the cost is very high.